File: fix-contaminant-breadcrumbs.js
#!/usr/bin/env node
/**
 * Fix contaminant breadcrumb URLs
 * 
 * Issues to fix:
 * 1. Double "contamination" suffix in URLs
 * 2. Missing subcategory level in URL structure
 * 
 * Correct pattern: /contamination/{category}/{subcategory}/{slug}
 */

const fs = require('fs');
const path = require('path');
const yaml = require('yaml');

const contaminantsDir = path.join(__dirname, 'frontmatter', 'contaminants');
const files = fs.readdirSync(contaminantsDir).filter(f => f.endsWith('.yaml'));

console.log(`Found ${files.length} contaminant files to process\n`);

let fixedCount = 0;
let errorCount = 0;

files.forEach(file => {
  const filePath = path.join(contaminantsDir, file);
  
  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const data = yaml.parse(content);
    
    if (!data?.metadata) {
      console.log(`⚠️  Skipping ${file}: No metadata found`);
      return;
    }
    
    const { category, subcategory, slug } = data.metadata;
    
    if (!category || !subcategory || !slug) {
      console.log(`⚠️  Skipping ${file}: Missing category, subcategory, or slug`);
      return;
    }
    
    // Build correct URL
    const correctUrl = `/contamination/${category}/${subcategory}/${slug}`;
    
    // Check if breadcrumb exists and needs fixing
    if (data.metadata.breadcrumb && Array.isArray(data.metadata.breadcrumb)) {
      const lastBreadcrumb = data.metadata.breadcrumb[data.metadata.breadcrumb.length - 1];
      
      if (lastBreadcrumb && lastBreadcrumb.href !== correctUrl) {
        const oldUrl = lastBreadcrumb.href;
        lastBreadcrumb.href = correctUrl;
        
        // Write back to file
        const newContent = yaml.stringify(data);
        fs.writeFileSync(filePath, newContent, 'utf8');
        
        console.log(`✅ Fixed ${file}`);
        console.log(`   Old: ${oldUrl}`);
        console.log(`   New: ${correctUrl}\n`);
        
        fixedCount++;
      }
    }
  } catch (error) {
    console.error(`❌ Error processing ${file}:`, error.message);
    errorCount++;
  }
});

console.log(`\n${'='.repeat(60)}`);
console.log(`✅ Fixed: ${fixedCount} files`);
console.log(`❌ Errors: ${errorCount} files`);
console.log(`📊 Total: ${files.length} files`);
console.log(`${'='.repeat(60)}\n`);

File: fix-contaminant-breadcrumbs.py
#!/usr/bin/env python3
"""
Fix contaminant breadcrumb URLs

Issues to fix:
1. Double "contamination" suffix in URLs
2. Missing subcategory level in URL structure

Correct pattern: /contamination/{category}/{subcategory}/{slug}
"""

import os
import yaml
from pathlib import Path

contaminants_dir = Path("frontmatter/contaminants")
files = list(contaminants_dir.glob("*.yaml"))

print(f"Found {len(files)} contaminant files to process\n")

fixed_count = 0
error_count = 0

for file_path in files:
    try:
        with open(file_path, "r", encoding="utf-8") as f:
            data = yaml.safe_load(f)

        if not data or "metadata" not in data:
            print(f"⚠️  Skipping {file_path.name}: No metadata found")
            continue

        metadata = data["metadata"]
        category = metadata.get("category")
        subcategory = metadata.get("subcategory")
        slug = metadata.get("slug")

        if not all([category, subcategory, slug]):
            print(
                f"⚠️  Skipping {file_path.name}: Missing category, subcategory, or slug"
            )
            continue

        # Build correct URL
        correct_url = f"/contamination/{category}/{subcategory}/{slug}"

        # Check if breadcrumb exists and needs fixing
        if "breadcrumb" in metadata and isinstance(metadata["breadcrumb"], list):
            last_breadcrumb = metadata["breadcrumb"][-1]

            if last_breadcrumb and last_breadcrumb.get("href") != correct_url:
                old_url = last_breadcrumb["href"]
                last_breadcrumb["href"] = correct_url

                # Write back to file
                with open(file_path, "w", encoding="utf-8") as f:
                    yaml.dump(
                        data,
                        f,
                        default_flow_style=False,
                        allow_unicode=True,
                        sort_keys=False,
                    )

                print(f"✅ Fixed {file_path.name}")
                print(f"   Old: {old_url}")
                print(f"   New: {correct_url}\n")

                fixed_count += 1

    except Exception as error:
        print(f"❌ Error processing {file_path.name}: {error}")
        error_count += 1

print(f"\n{'=' * 60}")
print(f"✅ Fixed: {fixed_count} files")
print(f"❌ Errors: {error_count} files")
print(f"📊 Total: {len(files)} files")
print(f"{'=' * 60}\n")
